crates/iota/src/name_commands.rs

```diff
@@ -441,23 +449,29 @@ impl NameCommand {
                 new_address,
                 opts,
             } => {
-                let nft_id = get_owned_nft_by_name::<IotaNamesRegistration>(&domain, context)
-                    .await?
-                    .id();
+                let entry = get_registry_entry(&domain, &iota_client).await?;
+                let new_address =
+                    get_identity_address(new_address.map(KeyIdentity::Address), context)?;
+                if entry
+                    .name_record
+                    .target_address
+                    .is_some_and(|a| a == new_address)
+                {
+                    anyhow::bail!("target address is already set to the given value");
+                }
+                let nft = get_proxy_nft_by_name(&domain, context).await?;
                 let iota_names_config = get_iota_names_config(&iota_client).await?;
 
                 NameCommandResult::Client(
                     IotaClientCommands::Call {
-                        package: iota_names_config.package_address.into(),
-                        module: "controller".to_owned(),
+                        package: nft.package_id(&iota_client).await?,
+                        module: nft.module_name().to_owned(),
                         function: "set_target_address".to_owned(),
                         type_args: Default::default(),
                         args: vec![
                             IotaJsonValue::from_object_id(iota_names_config.object_id),
-                            IotaJsonValue::from_object_id(nft_id),
-                            IotaJsonValue::new(serde_json::to_value(
-                                new_address.into_iter().collect::<Vec<_>>(),
-                            )?)?,
+                            IotaJsonValue::from_object_id(nft.id()),
+                            IotaJsonValue::new(serde_json::to_value(new_address)?)?,
```

Comment:
I get this 
```
Could not serialize argument of type StructInstantiation(DatatypeHandleIndex(6), [Address]) at 2 into vector<address>. Got error: Cannot convert string arg 0xb8568d8c759dc5449d43fb1e05dbb3132c7d16dc145a0d127e72e30804834e18 to vector<address>
```

Reply:
oops, I meant to wrap it again and totally forgot 😆